### PR TITLE
Add support for testing agains multiple Rails versions

### PR DESCRIPTION
**Pull request by vr4b4c**
### Aim & Solution
Install and configure Gemika

### CODE REVIEW

File: rails.6.1.gemfile.lock [hunk 1]
```diff
@@ -0,0 +1,282 @@
...
+      jsonapi-serializer
+      jsonapi_parameters
+      pagy
+      rails
+      responders
+
+GEM
+  remote: https://rubygems.org/
+  specs:
+    actioncable (6.1.7)
```
Comment (uncoverd) [inline rails.6.1.gemfile.lock:16]: I'm not used to gem development, could you explain why we need a gemfile in this project? I was expecting just a gemspec

**Review by uncoverd — APPROVED** [review]

File: rails.6.1.gemfile.lock (hunk 1 above)
Comment (vr4b4c) [reply rails.6.1.gemfile.lock:16]: the primary reason is, that's how [gemika](https://github.com/makandra/gemika) works. If you want to test your code against different dependency setups, then you have to have one Gemfile per dependency setup

Comment (uncoverd) [reply rails.6.1.gemfile.lock:16]: And what about before, could we have gone without it if we locked rails in the gemspec?

Comment (vr4b4c) [reply rails.6.1.gemfile.lock:16]: Why do you want to get rid of Gemfile and Gemfile.lock?

> could we have gone without the Gemfile & Gemfile.lock if we locked rails in the gemspec instead?
Rails in gemspec is defined as a runtime dependency, if we lock it to a certain version, that would mean our gem is working with only one version of Rails

Comment (uncoverd) [reply rails.6.1.gemfile.lock:16]: I was taking a look at jsonapi_query_builder and this gem yesterday and noticed that they're setup differently, one has everything in the gemspec https://github.com/infinum/jsonapi-query_builder/blob/master/jsonapi-query_builder.gemspec, while this one had gems specified in both the gemfile and gemspec

> Rails in gemspec is defined as a runtime dependency, if we lock it to a certain version, that would mean our gem is compatible with only one version of Rails.

Agreed, to be more specific I wanted to ask "why do we need the gemfile.lock, couldn't we lock the rails version to >5, < 8 in the gemspec instead?"

I'll take some time and re-read https://yehudakatz.com/2010/12/16/clarifying-the-roles-of-the-gemspec-and-gemfile/, I'm not entirely clear on the purpose and suggested usages of these files

Comment (vr4b4c) [reply rails.6.1.gemfile.lock:16]: 👍 from my perspective, I don't think there is a practical difference between specifying development deps in gemspec vs Gemfile

**Pull request merged**